distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == schema2.MediaTypePluginConfig {
```

Comment:
I tried this approach before blacklisting plugins. The problem is that docker pull official Hub images such busybox, python, ubuntu has mediatype set to  "application/octet-stream" and not the expected schema2.MediaTypeConfig ("application/vnd.docker.container.image.v1+json").

https://github.com/docker/distribution/issues/1621 seems relevant.